lib/galaxy/webapps/galaxy/services/history_contents.py

```diff
@@ -939,6 +940,18 @@ def __index_v2(
         Allows additional filtering of contents and custom serialization.
         """
         history = self._get_history(trans, history_id)
+
+        # TODO: Would be better if the JobConnections Manager could be used in
+        # history_contents manager instead, can't now because of circular import there
```

Comment:
You could move the `JobConnectionsManager` out of `jobs.py` and write it into e.g. `jobs_connections.py` and then import it without triggering a circular dependency. Alternatively you could just leave it as it is and rather replace the lines below with a single call to the `JobConnectionsManager` e.g. something like: `filter_query_params = self.job_connections_manager.parse_query_filters(history_id, filter_query_params)`.

Reply:
Done!